### PR TITLE
Report complete

### PR DESCRIPTION
**Pull request by damianlluch**
Added the module of people with complete data that will be approved as members, whether student, collaborator or active member.
Check that the conditions are mutually exclusive.

### CODE REVIEW

**Comment by eduzen** [comment]
El codigo de `ReportMissing` es muy parecido me parece que se puede heredar de ahi. Después no me quedo claro que es lo que este reporte agrega y algun test no vendria mal.

File: website/members/views.py [hunk 1]
```diff
@@ -321,6 +321,162 @@ def get(self, request):
...
         return render(request, 'members/report_missing.html', context)
 
+class ReportComplete(View):
+    """Handles the report on people who are in a position to be approved as members"""
+
+    MAIL_SUBJECT = "Continuación del trámite de inscripción a la Asociación Civil Python Argentina"
+    MAIL_FROM = 'Lalita <lalita@ac.python.org.ar>'
+    MAIL_MANAGER = 'presidencia@ac.python.org.ar>'
+
+    def _generate_letter(self, member):
```
Comment (facundobatista) [inline website/members/views.py]: If the person is ready to be a member, doesn't need to generate a letter

File: website/members/views.py [hunk 2]
```diff
@@ -321,6 +321,162 @@ def get(self, request):
...
+        missing_payment = member.first_payment_month is not None and member.category.fee != 0
+        missing_signed_letter = member.has_subscription_letter
+        missing = missing_signed_letter and missing_nickname and picture
+
+        return {
+            'missing': missing,
+            'missing_student_certif': missing_student_certif,
+            'missing_payment': missing_payment,
+            'missing_collab_accept': missing_collab_accept,
+        }
```
Comment (facundobatista) [inline website/members/views.py]: No point in returning a complex structure here, all it needs to know if the member has everything complete and ready to be approved, or not. A bool is enough.

File: website/members/views.py [hunk 3]
```diff
@@ -321,6 +321,162 @@ def get(self, request):
...
+                info[k] = "COMPLETO"
+            else:
+                if info['missing'] and info['missing_payment']:
+                    info[k] = "COMPLETO"
+                else:
+                    if info['missing'] and info['missing_collab_accept']:
+                        info[k] = "COMPLETO"
+                    else:
+                        info[k] = "INCOMPLETO"
+
```
Comment (facundobatista) [inline website/members/views.py]: All this is superfluous. `_analyze_member` should be called something like `_is_member_complete` and return a bool.

File: website/members/templates/members/report_complete.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                    <button id="selected" type="submit">Mandar</button>
+                <table class="table">
+                <tr> 
+                    <th>Nombre</th>
+                    <th>Mail</th>
+                    <th>¿Mandar mail?</th>
+                </tr>
+                {% for item in completes %}
+                    <tr> 
+                        <td>{{ item.member.entity.full_name }}</td>
```
Comment (facundobatista) [inline website/members/templates/members/report_complete.html]: We need to show the DNI too, here.

File: website/members/views.py [hunk 5]
```diff
@@ -321,6 +321,162 @@ def get(self, request):
...
+        errors_code = str(uuid.uuid4())
+        for member_id in to_send_mail_ids:
+            member = Member.objects.get(id=member_id)
+
+            # create the mail text from the template
+            missing_info = self._analyze_member(member)
+            missing_info['annual_fee'] = member.category.fee * 12
+            missing_info['member'] = member
+            missing_info['on_purpose_missing_var'] = "ERROR"
+            text = render_to_string('members/mail_missing.txt', missing_info)
```
Comment (facundobatista) [inline website/members/views.py]: Why are you sending a "missing" mail here?

**Comment by facundobatista** [comment]
Este entró al final en https://github.com/PyAr/asoc_members/pull/92 (pero con todos los commits con el autor correspondiente).

Gracias!!

**Pull request closed without merging**